docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx

```diff
@@ -0,0 +1,97 @@
+---
+description: How to get the L2 NFTs owned by an account.
+image: /img/logo/WASP_logo_dark.png
+tags:
+  - EVM
+  - how-to
+  - native tokens
+  - L2 NFTs
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+
+# Get NFTs Owned by an Account
+
+This guide will show you how to get the L2 NFTs owned by an account using the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md#getl2nfts) function from the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md) [magic contract](../../../reference/magic-contract/introduction.md).
+
+<GetNftMetadata />
+
+## Understanding the `getL2NFTs` Function
+
+The `getL2NFTs` function is a view function provided by the `ISCAccounts` interface. It takes an `ISCAgentID` as an argument and returns an array of `NFTID` objects. This function allows users to query and retrieve the list of L2 NFTs owned by a specified agent. The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. The returned array contains the identifiers of the NFTs, which can then be used to fetch more details about each NFT if needed.
+
+## Why Would a User Need to Retrieve Owned NFTs?
+
+Retrieving the NFTs owned by a user can serve several purposes:
+
+- ### Portfolio Management
+
+  - Users can manage their digital assets effectively by viewing all their NFTs in one place.
+
+- ### Marketplace Listing
+
+  - Users can list their NFTs on various marketplaces by verifying their ownership.
+
+- ### Collection Display
+
+  - Users can showcase their NFT collections on social media, personal websites, or within specific applications.
+
+- ### Trading and Transfers
+
+  - Knowing the NFTs owned helps users decide which assets to trade or transfer to other users.
+
+- ### Application Integration
+
+  - Applications can provide customized experiences based on the NFTs a user owns, such as granting access to special content or features.
```

Comment:
```suggestion
```

I would keep the how-tos on point. Not too much stuff around it. We can create user stories in tutorials imo

Reply:
Ok, removed.